crates/matrix-sdk-base/src/store/memory_store.rs

```diff
@@ -608,6 +609,14 @@ impl StateStore for MemoryStore {
     async fn remove_room(&self, room_id: &RoomId) -> Result<()> {
         self.remove_room(room_id).await
     }
+
+    async fn room_timeline(
+        &self,
+        _room_id: &RoomId,
+    ) -> Result<Option<(BoxStream<'static, Result<SyncRoomEvent>>, Option<String>)>> {
+        // The `MemoryStore` doesn't cache any events
+        Ok(None)
+    }
```

Comment:
Since the `MemoryStore` stores all events in memory, I thought it would make sense not to store the timeline or we would fill the ram fairly quickly.

Reply:
That's very counter intuitive though - if I had a store, I'd expect them to behave roughly the same in terms of features. Like, with the _faster_ memory store, we'd actually be slower in showing timelines as they are fetched from remote on every call. 

If I have a large data set and use the memory store, I expect that people are aware that we'd have to consume their RAM to store stuff. We could make a note about that on the docs or allow the creator of the `MemoryStore` to configure it so that it _doesn't_ store timelines ( #498 will make that easier for users), but it must generally support it, I'd say.

Reply:
This concern was raised by @poljar . But I'm fine with adding also an implementation for MemoryStore, and IndexedDBStore.

Reply:
I added now an implementation for the `MemoryStore`. Tomorrow I will have a look at the `IndexDBStore`